apps/api/v1/pages/api/event-types/[id]/_delete.ts

```diff
@@ -48,10 +51,32 @@ export async function deleteHandler(req: NextApiRequest) {
 async function checkPermissions(req: NextApiRequest) {
   const { userId, isSystemWideAdmin } = req;
   const { id } = schemaQueryIdParseInt.parse(req.query);
+
   if (isSystemWideAdmin) return;
-  /** Only event type owners can delete it */
-  const eventType = await prisma.eventType.findFirst({ where: { id, userId } });
-  if (!eventType) throw new HttpError({ statusCode: 403, message: "Forbidden" });
+
+  const eventType = await prisma.eventType.findFirst({ where: { id } });
+  if (!eventType) throw new HttpError({ statusCode: 404, message: `Event type with ID ${id} not found` });
+
+  if (eventType.teamId) {
+    /** Only team owners can delete team events */
+    await checkTeamEventEditPermission(req, { ...eventType, userId: eventType.userId ?? undefined });
+  } else {
+    /** Only parent team owners can remove user assignment */
+    if (eventType.parentId) {
+      const reqWithAdditionalInfo = {
+        ...req,
+        body: {
+          ...req.body,
+          parentId: eventType.parentId,
+        },
+        userId,
+      } as NextApiRequest;
+      return await checkParentEventOwnership(reqWithAdditionalInfo);
```

Comment:
Workaround otherwise would need to make changes in `checkParentEventOwnership` and a lot of other places